### PR TITLE
Updating user model to support parent email notifications

### PR DESCRIPTION
**Pull request by daynew**
We are adding a new feature to enable parents to get notification emails
about their child's progress. This changes make it so when a student
account changes, it will also update the parent's email preferences.

## Future work
Still need to test integration with front end. But that will be done in a futre PR.

## Links
- [jira](https://codedotorg.atlassian.net/browse/FND-1093)

## Testing story
* Added unit tests.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/user.rb [hunk 1]
```diff
@@ -223,6 +225,19 @@ def save_email_preference
...
 
+  # Enables/disables email notifications for the parent.
+  def save_parent_email_preference
+    if student? && parent_email.present?
+      EmailPreference.upsert!(
+        email: parent_email,
+        opt_in: parent_email_preference_opt_in.downcase == "yes",
+        ip_address: parent_email_preference_request_ip,
+        source: parent_email_preference_source,
+        form_kind: nil
```
Comment (bethanyaconnor) [inline dashboard/app/models/user.rb:236]: Do we want to double check that this isn't used?

File: dashboard/test/models/user_test.rb [hunk 2]
```diff
@@ -2199,6 +2205,77 @@ def email_preference_params(**args)
...
+    # There should be no email preference because no parent_email was defined.
+    assert_nil email_preference
+    user.update!(parent_email_params)
+    email_preference = EmailPreference.find_by_email(parent_email_params[:parent_email])
+    # There should now be an email preference because the user was updated with a parent_email.
+    assert_not_nil email_preference
+    assert_parent_email_params_equals_email_preference parent_email_params, email_preference
+  end
+
+  def assert_parent_email_params_equals_email_preference(parent_email_params, email_preference)
```
Comment (bethanyaconnor) [inline dashboard/test/models/user_test.rb]: nit: should this be above all of the tests that use it?

**Review by bethanyaconnor — APPROVED** [review]
🎉

File: dashboard/app/models/user.rb (hunk 1 above)
Comment (daynew) [reply dashboard/app/models/user.rb:236]: I just did a quick pass through all the code mentioning "form_kind". It doesn't look like anything which would affect us. There are also unit tests which validate that its okay to have `nil` as a value

**Pull request merged**